source/frontend/src/features/documents/list/DocumentResults/DocumentResultsColumns.tsx

```diff
@@ -112,46 +132,103 @@ const renderActions = (
 ) => {
   return function ({ row: { original, index } }: CellProps<DocumentRow, string>) {
     const { hasClaim } = useKeycloakWrapper();
-    return (
-      <StyledIconsRow className="no-gutters">
-        {hasClaim(Claims.DOCUMENT_VIEW) && (
-          <Col>
-            <Button
-              data-testid="document-view-button"
-              icon={<FaEye size={24} id={`document-view-${index}`} title="document view details" />}
-              onClick={() => original?.id && onViewDetails(DocumentRow.toApi(original))}
-            ></Button>
-          </Col>
-        )}
-        {hasClaim(Claims.DOCUMENT_DELETE) && (
-          <Col>
+
+    const documentInError =
+      (original.mayanDocumentId === null || original.mayanDocumentId === -1) &&
+      original.queueStatusTypeCode &&
+      (original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.PIMS_ERROR ||
+        original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.MAYAN_ERROR);
+
+    const documentProcessing =
+      (original.mayanDocumentId === null || original.mayanDocumentId < 1) &&
+      original.queueStatusTypeCode &&
+      (original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.PENDING ||
+        original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.PROCESSING);
+
+    const canViewDocument =
+      (original.mayanDocumentId &&
```

Comment:
the mayan document id being present isn't sufficient to download a document if it is still being processed - need to check for success.

Reply:
I wouldn't expect a MayanId if it's still being processed, is this not the case?

Reply:
Mayan ids are returned from mayan immediately so that you can track the progress of the document as it is processed. documents may have mayan ids but no file contents while they are being processed.

Reply:
UIpdated